mlflow/deployments/mlflow/__init__.py

```diff
@@ -6,6 +6,12 @@ class MLflowDeploymentClient(BaseDeploymentClient):
     TODO
     """
 
+    def __init__(self, target_uri):
+        from mlflow.gateway.client import MlflowGatewayClient
+
+        super().__init__(target_uri)
+        self.client = MlflowGatewayClient(target_uri)
```

Comment:
Since we're deprecating this client, will users see a deprecation warning every time they call this API?

Reply:
Good point. Two

1. Suppress the warning during import using `warnings.catch_warnings`.
2. Isolate `MlflowGatewayClient` (e.g. in `mlflow/deployments/mlflow/client.py`)

PR to add a deprecation warning: https://github.com/mlflow/mlflow/pull/10460

Reply:
Option 3: Reimplement the APIs in `MlflowDeploymentClient`, since the API URLs are changing: https://github.com/mlflow/mlflow/pull/10466. I prefer this

Reply:
Sounds good, updating